### PR TITLE
core: fix too drastic continuity test on speed envelopes

### LINKED ISSUE #7901: core: Assert check failed
### What happened?

The following error is happening sometimes on BASIC timetables:
```
{"status":500,"type":"core:assert_error","context":{"url":"http://osrd-rec-core.osd-rec01.svc.cluster.local/v2/standalone_simulation","file_location":"AbstractAllowanceWithRanges.java:137","assert_message":"Discontinuity on the edges of the allowance region","stack_trace":["AbstractAllowanceWithRanges.java:137","StandaloneSimulation.kt:259","StandaloneSimulation.kt:102","SimulationEndpoint.kt:56","FkRegex.java:153","FkRegex.java:217","FkChain.java:72","TkFork.java:98","TkDataDog.kt:27","TkDataDog.kt:16","TkDataDog.kt:15","TkWrap.java:58","TkFallback.java:84","TkFallback.java:66","TkWrap.java:58","TkSlf4j.java:110","BkBasic.java:123","BkBasic.java:99","BkSafe.java:46","BkWrap.java:51","BkParallel.java:81","ThreadPoolExecutor.java:1144","ThreadPoolExecutor.java:642","Thread.java:1583"]},"message":"assert check failed"}
```

### What did you expect to happen?

Have a "normal" error. This is blocking scenario opening.

### How can we reproduce it (as minimally and precisely as possible)?

* Open a BASIC timetable and look for the network in the navigator : https://rec-osrd.reseau.sncf.fr/operational-studies/projects/4/studies/50/scenarios/131

* Or locally, using the last dump on `France` infra, and provide the given payload (provided aside) to `http://localhost:4000/api/v2/timetable/<scenario>/train_schedule/`

### What operating system, browser and environment are you using?

- **Browser**: Firefox 

### OSRD version (top right corner `Account` button > `Informations`)

435263b

### PR DESCRIPTION
**Pull request by bougue-pe**
This case can happen after rounding errors when allowance is processed.

fix #7901

### CODE REVIEW

File: core/envelope-sim/src/test/java/fr/sncf/osrd/envelope/EnvelopeBuilderTest.java [hunk 1]
```diff
@@ -14,6 +14,16 @@ public void testEnvelopeBuilder() {
         assertTrue(env.continuous);
     }
 
+    @Test
+    public void testEnvelopeBuilderEpsilonSpeedDiscontinuity() {
```
Comment (Erashin) [inline core/envelope-sim/src/test/java/fr/sncf/osrd/envelope/EnvelopeBuilderTest.java]: You're testing the continuity here: `testEnvelopeBuilderEpsilonSpeedContinuity`? And I'd add another test (or test case if you wanna go for a parametrised test) to show the discontinuity as soon as the difference is equal to 1e-5.

**Review by Erashin — APPROVED** [review]
LGTM

File: core/envelope-sim/src/test/java/fr/sncf/osrd/envelope/EnvelopeBuilderTest.java (hunk 1 above)
Comment (bougue-pe) [reply core/envelope-sim/src/test/java/fr/sncf/osrd/envelope/EnvelopeBuilderTest.java]: Thanks, you're right.
Done (amended commit): https://github.com/OpenRailAssociation/osrd/compare/9f35aa419bfef92a3946061fd03defbd8aeee5fc..1310a4d443341c9da08f67318b22e3f5eb856422

**Review by Khoyo — APPROVED** [review]
LGTM

**Pull request merged**